code/modules/mob/living/living_defense.dm

```diff
@@ -387,6 +387,17 @@
 
 /mob/living/emp_act(severity)
 	. = ..()
+
+	// Damage robotic simplemobs and basicmobs when EMPd
+	if((istype(src, /mob/living/simple_animal) || istype(src, /mob/living/basic)) && mob_biotypes & MOB_ROBOTIC)
+		switch (severity)
+			if (EMP_LIGHT)
+				visible_message(span_danger("[src] shakes violently, its parts coming loose!"))
+				apply_damage(maxHealth * 0.6)
+			if (EMP_HEAVY)
+				visible_message(span_danger("[src] suddenly bursts apart!"))
+				apply_damage(maxHealth)
```

Comment:
This will qdelete the mob, right? does that mean any parent calls of emp_act that do things with it will end up hanging references that will hard delete?

Reply:
Tested emping a bunch of different mobs and didn't get a single related hard del in the del() in debug menu so I think it's fine

Reply:
how long did you wait?

Reply:
about 40 minutes

Reply:
![image](https://user-images.githubusercontent.com/22856555/154870898-bd84cf71-88e5-45ce-9c01-eaa11fba48be.png)